Linker errors in `build x86_64-linux-gnummtk`:  "undefined reference to `jl_gc_debug_fprint_critical_error"

The missing function was recently introduced in PR #54835 by @topolarity .

Maybe people working on MMTK support like @d-netto or @udesou will quickly spot what the issue is?

See e.g. [this CI log](https://buildkite.com/julialang/julia-master/builds/50767/steps/canvas?sid=019976b2-df71-4956-b017-6dc8ee72c01d)

```
/usr/local/bin/../lib/gcc/x86_64-linux-gnu/9.1.0/../../../../x86_64-linux-gnu/bin/ld: ./builtins.o: in function `jl_f_intrinsic_call':
[buildroot]/src/builtins.c:2459: undefined reference to `jl_gc_debug_fprint_critical_error'
/usr/local/bin/../lib/gcc/x86_64-linux-gnu/9.1.0/../../../../x86_64-linux-gnu/bin/ld: ./signal-handling.o: in function `jl_fprint_critical_error':
[buildroot]/src/signal-handling.c:652: undefined reference to `jl_gc_debug_fprint_status'
/usr/local/bin/../lib/gcc/x86_64-linux-gnu/9.1.0/../../../../x86_64-linux-gnu/bin/ld: [buildroot]/src/signal-handling.c:653: undefined reference to `jl_gc_debug_fprint_critical_error'
/usr/local/bin/../lib/gcc/x86_64-linux-gnu/9.1.0/../../../../x86_64-linux-gnu/bin/ld: ./task.o: in function `jl_finish_task':
[buildroot]/src/task.c:355: undefined reference to `jl_gc_debug_fprint_critical_error'
/usr/local/bin/../lib/gcc/x86_64-linux-gnu/9.1.0/../../../../x86_64-linux-gnu/bin/ld: ./safepoint.o: in function `jl_safepoint_init':
[buildroot]/src/safepoint.c:109: undefined reference to `jl_gc_debug_fprint_critical_error'
/usr/local/bin/../lib/gcc/x86_64-linux-gnu/9.1.0/../../../../x86_64-linux-gnu/bin/ld: ./rtutils.o: in function `__stack_chk_fail':
[buildroot]/src/rtutils.c:246: undefined reference to `jl_gc_debug_fprint_critical_error'
/usr/local/bin/../lib/gcc/x86_64-linux-gnu/9.1.0/../../../../x86_64-linux-gnu/bin/ld: [buildroot]/usr/lib/libjulia-internal.so.1.13.0: hidden symbol `jl_gc_debug_fprint_status' isn't defined
/usr/local/bin/../lib/gcc/x86_64-linux-gnu/9.1.0/../../../../x86_64-linux-gnu/bin/ld: final link failed: bad value
collect2: error: ld returned 1 exit status
make[1]: *** [Makefile:449: [buildroot]/usr/lib/libjulia-internal.so.1.13.0] Error 1
```

--- a/src/gc-debug.c
+++ b/src/gc-debug.c
@@ -470,8 +470,8 @@ void jl_gc_debug_fprint_status(ios_t *s) JL_NOTSAFEPOINT
     uint64_t pool_count = jl_gc_debug_env.pool.num;
     uint64_t other_count = jl_gc_debug_env.other.num;
     jl_safe_fprintf(s, "Allocations: %" PRIu64 " "
-                   "(Pool: %" PRIu64 "; Other: %" PRIu64 "); GC: %d\n",
-                   pool_count + other_count, pool_count, other_count, gc_num.pause);
+                    "(Pool: %" PRIu64 "; Other: %" PRIu64 "); GC: %d\n",
+                    pool_count + other_count, pool_count, other_count, gc_num.pause);
 }
 
 void jl_gc_debug_fprint_critical_error(ios_t *s) JL_NOTSAFEPOINT

--- a/src/gc-mmtk.c
+++ b/src/gc-mmtk.c
@@ -1172,7 +1172,7 @@ JL_DLLEXPORT jl_taggedvalue_t *jl_gc_find_taggedvalue_pool(char *p, size_t *osiz
     return NULL;
 }
 
-void jl_gc_debug_critical_error(void) JL_NOTSAFEPOINT
+void jl_gc_debug_fprint_critical_error(ios_t *s) JL_NOTSAFEPOINT
 {
 }
 
@@ -1181,14 +1181,14 @@ int gc_is_collector_thread(int tid) JL_NOTSAFEPOINT
     return 0;
 }
 
-void jl_gc_debug_print_status(void) JL_NOTSAFEPOINT
+void jl_gc_debug_fprint_status(ios_t *s) JL_NOTSAFEPOINT
 {
     // May not be accurate but should be helpful enough
     uint64_t pool_count = gc_num.poolalloc;
     uint64_t big_count = gc_num.bigalloc;
-    jl_safe_printf("Allocations: %" PRIu64 " "
-                   "(Pool: %" PRIu64 "; Big: %" PRIu64 "); GC: %d\n",
-                   pool_count + big_count, pool_count, big_count, gc_num.pause);
+    jl_safe_fprintf(s, "Allocations: %" PRIu64 " "
+                    "(Pool: %" PRIu64 "; Big: %" PRIu64 "); GC: %d\n",
+                    pool_count + big_count, pool_count, big_count, gc_num.pause);
 }
 
 JL_DLLEXPORT size_t jl_gc_external_obj_hdr_size(void)

